internal/backend/db/dbgorm/connections.go

```diff
@@ -21,12 +22,30 @@ func (gdb *gormdb) createConnection(ctx context.Context, conn *scheme.Connection
 	return gdb.createEntityWithOwnership(ctx, createFunc, conn, conn.ProjectID)
 }
 
+func (gdb *gormdb) deleteConnectionsAndVars(where string, args ...interface{}) error {
+	// should be transactional with context already applied
+
+	var ids []sdktypes.UUID
+	q := gdb.db.Model(&scheme.Connection{})
+	q = q.Clauses(clause.Returning{Columns: []clause.Column{{Name: "connection_id"}}})
+	if err := q.Delete(&ids, where, args).Error; err != nil {
+		return err
+		// REVIEW: proceed to vars deletion if there are any?
+	}
+
+	if len(ids) > 0 {
+		return gdb.db.Where("var_id IN (?)", ids).Delete(&scheme.Var{}).Error
```

Comment:
delete from 2 different schemas? I don't think so. 
maybe it's possible with assocications, but varID isn't a foreign key since it could be wither endID or connectionID. moreover gorm cannot cascade on delete from soft-deleted entries (e.g. env or connection)

Reply:
I think there is a way to do polymorphic foreign keys, probably over kill for this case
but we should probably wrap it in transaction if it is mandatory to be in transaction (it's okay to have tx in tx)

Reply:
I already dealt with this problem while implementing users. There is no way in gorm to enforce polymorphic foreign keys unfortunately.
regarding transaction - it's an internal method which is called in transaction already. I don't want to add overhead of recursive transactions. I tested this while implementing users and got to the conclusion that I will use internal methods and single transaction where possible. Recursive transactions is uneeded overhead IMO for now